elasticdl/python/common/k8s_job_monitor.py

```diff
@@ -43,24 +44,28 @@ def monitor_status(self):
         pod_succeeded = False
 
         while True:
-            pod = self.client.get_pod(self.pod_name)
-            if pod is None:
-                retry_num += 1
-                if retry_num > MAX_READ_POD_RETRIES:
-                    logger.error("{} Not Found".format(self.pod_name))
+            try:
+                pod = self.client.get_pod(self.pod_name)
+                if pod is None:
+                    retry_num += 1
+                    if retry_num > MAX_READ_POD_RETRIES:
+                        logger.error("{} Not Found".format(self.pod_name))
+                        break
+                    time.sleep(10)
+                    continue
+
+                logger.info("Pod Status : %s" % pod.status.phase)
+                if pod.status.phase == PodStatus.SUCCEEDED:
+                    pod_succeeded = True
                     break
-                time.sleep(10)
-                continue
-
-            logger.info("Pod Status : %s" % pod.status.phase)
-            if pod.status.phase == PodStatus.SUCCEEDED:
-                pod_succeeded = True
-                break
-            elif pod.status.phase == PodStatus.FAILED:
-                logger.info(self.client.get_pod_log(self.pod_name))
-                break
-            else:
-                time.sleep(30)
+                elif pod.status.phase == PodStatus.FAILED:
+                    logger.info(self.client.get_pod_log(self.pod_name))
+                    break
+                else:
+                    time.sleep(30)
+            except Exception:
+                traceback.print_exc()
```

Comment:
Should we record the trace in the log?

Reply:
The log is unused for users, so I will remove the code.